.golangci.yml

```diff
@@ -159,30 +159,110 @@ linters-settings:
+  - gosimple                  # Linter for Go source code that specializes in simplifying code.
+  - gosmopolitan              # Report certain i18n/l10n anti-patterns in your Go codebase.
+  - govet                     # Vet examines Go source code and reports suspicious constructs, such as Printf calls whose arguments do not align with the format string
+  - grouper                   # Analyze expression groups.
+  - importas                  # Enforces consistent import aliases.
+  - ineffassign               # Detects when assignments to existing variables are not used
+  - loggercheck               # Checks key value pairs for common logger libraries (kitlog,klog,logr,zap).
+  - mirror                    # Reports wrong mirror patterns of bytes/strings usage.
+  - misspell                  # Finds commonly misspelled English words in comments.
+  - nakedret                  # Checks that functions with naked returns are not longer than a maximum size (can be zero).
+  - nilerr                    # Finds the code that returns nil even if it checks that the error is not nil.
+  - nosprintfhostport         # Checks for misuse of Sprintf to construct a host with port in a URL.
+  - perfsprint                # Checks that fmt.Sprintf can be replaced with a faster alternative.
+  - prealloc                  # Finds slice declarations that could potentially be pre-allocated
+  - reassign                  # Checks that package variables are not reassigned.
+  - revive                    # Fast, configurable, extensible, flexible, and beautiful linter for Go. Drop-in replacement of golint.
+  - sloglint                  # Ensure consistent code style when using log/slog.
+  - staticcheck               # It's a set of rules from staticcheck. It's not the same thing as the staticcheck binary. The author of staticcheck doesn't support or approve the use of staticcheck as a library inside golangci-lint.
+  - testableexamples          # Linter checks if examples are testable (have an expected output).
+  - tparallel                 # tparallel detects inappropriate usage of t.Parallel() method in your Go test codes
+  - unconvert                 # Remove unnecessary type conversions
+  - unparam                   # Reports unused function parameters
+  - unused                    # Checks Go code for unused constants, variables, functions and types
+  - whitespace                # Tool for detection of leading and trailing whitespace.
+  - zerologlint               # Detects the wrong usage of zerolog that a user forgets to dispatch with Send or Msg.
+
+    # TODO: Requires some code changes before enabling:
+
+# - bodyclose         # Checks whether HTTP response body is closed successfully.
+# - contextcheck      # Check whether the function uses a non-inherited context.
+# - cyclop            # Checks function and package cyclomatic complexity.
+# - dupl              # Tool for code clone detection.
```

Comment:
problem I have with code duplication: not all duplication is an issue. Only if the same concept is duplicated but not simply duplication of boilerplate code...

Reply:
Correct, this one only complains about our tests, and it;s okey to copy paste some stuff during testing phase, so I think we need may be configure it to ignore tests and enable.